sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -182,17 +192,21 @@ AZ_NODISCARD AZ_INLINE az_result az_iot_pnp_client_get_sas_password(
  *
  * @param[in] client The #az_iot_pnp_client to use for this call.
  * @param[in] component_name An #az_span specifying the component name to publish telemetry on.
- * @param[in] mqtt_topic An empty #az_span with sufficient capacity to hold the MQTT topic.
+ * @param[out] mqtt_topic A char buffer with sufficient capacity to hold the MQTT topic
+ * filter. On success, will be `NULL` terminated.
+ * @param[in] mqtt_topic_size The size of the passed buffer. Must be greater than 0.
  * @param[in] reserved Reserved for future use.  Must be NULL.
- * @param[out] out_mqtt_topic The output #az_span containing the MQTT topic.
+ * @param[out] out_mqtt_topic_length The optional output length of the mqtt topic filter. Can
+ * be `NULL`.
  * @return #az_result
  */
 AZ_NODISCARD az_result az_iot_pnp_client_telemetry_get_publish_topic(
     az_iot_pnp_client const* client,
     az_span component_name,
-    az_span mqtt_topic,
+    char* mqtt_topic,
+    int32_t mqtt_topic_size,
     void* reserved,
-    az_span* out_mqtt_topic);
+    int32_t* out_mqtt_topic_length);
 
```

Comment:
Actually.... I don't think we need a full on [Yalta Conference](https://en.wikipedia.org/wiki/Yalta_Conference) (as one of my buddies used to call overly drawn out philosophy meetings) on this :).  If you think this change is right, as I do, then just make it and see if we get feedback on next PR round.  It's not hard to change around if needed.

Reply:
Yea I think it should be change around. I'll make it before the `char*` and length stuff. So the last three will be the `char*`, size, and out_length

Reply:
@danewalton this is marked as resolved but I can't see the change. Unresolving to make this visible in the main discussion.

(I do agree we don't need to meet since the only requirements I got were around the 3 params. These need to be grouped for obvious reasons but the reserved can be in any position.)